cmake/fw_zip.cmake

```diff
@@ -0,0 +1,31 @@
+
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+function(generate_dfu_zip zip_out bin_file_list deps)
```

Comment:
Why have a `add_custom_target` for this (and hence a function) ?

Is this a target that user is expected to execute by his own ?

Have you tried to simply use `add_custom_command()`

Reply:
I'm not sure I understand what you mean. I need to use the add_custom_target in order to have the dependencies exposed outside this file (as with every other add_custom_command -> add_custom_target pair).

The target is added to the `ALL` target, so the user does not have to do anything to execute this.

Reply:
What I mean, is that you can have the `add_custom_command` as a post-build command on another target, and in that case it will not need an independent target name.

And even though the custom target is added to the `ALL`, then it will still show up as a target when you do `ninja <TAB>`

Reply:
The advantages of `POST_BUILD` on another target is that you remove the need for setting up an explicit dependency, as it will automatically execute the `POST_BUILD` step when the target builds (the target is the deps).

As well as for readers, it becomes much clearer that this is something the build command will do as part of building `<target>`.

Reply:
Great info, sounds like a better solution. I will try it out :)

Reply:
Due to the limitation of the target having to be defined in the same directory I could not get away from having a TARGET argument for the funtion. But all in all it looks alot better now! Note that I also had to add an explicit dependency between the s0 and s1 targets to guarantee that both s0 and s1 bin files are available when the zip is generated.

Reply:
Looks much better.
My eyes are much more relaxed now, thanks for this :)